### PR TITLE
update windows build guide

### LINKED ISSUE #47483: Update how to build PyTorch with CUDA Windows instructions
PyTorch currently could not be build using recommended `14.11.25503` minimal toolchain, see:
https://github.com/pytorch/pytorch/blame/b4b0fa637178baf9147416b550c7db70de6a5fa3/README.md#L258

But if one tries to following this instructions using PyTorch-1.7 or newer it will fail with as shown in:
https://github.com/pytorch/pytorch/issues/46208#issuecomment-707352250

cc @malfet @seemethere @walterddr @jlin27 @mruberry @peterjc123 @maxluk @nbcsm @guyang3532 @gunandrose4u @smartcat2010 @mszhanyi

### PR DESCRIPTION
**Pull request by mszhanyi**
Fixes #47483

### CODE REVIEW

**Comment by mszhanyi** [comment]
@pytorch/buildbot please rebuild it

File: README.md
Comment (walterddr) [inline README.md:254]: Is there a suggested python version for this matrix as well?

File: README.md
Comment (walterddr) [inline README.md:258]:
```suggestion
You can refer to [build_pytorch.bat](https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/win-test-helpers/build_pytorch.bat) script for environment variable configurations.
```
Also I am not sure what this paragraph has to do with the previous one. (might be in the wrong place?)

**Review by walterddr — CHANGES_REQUESTED** [review]
looks good overall. some wording might need a bit clarification. Please kindly take a look

File: README.md
Comment (walterddr) [inline README.md:256]:
```suggestion
Additional libraries are often needed such as: [Magma](https://developer.nvidia.com/magma), [oneDNN, a.k.a MKLDNN or DNNL](https://github.com/oneapi-src/oneDNN) and [Sccache](https://github.com/oneapi-src/oneDNN). Please refer to the [installation-helper](https://github.com/pytorch/pytorch/tree/master/.jenkins/pytorch/win-test-helpers/installation-helpers) script to install them.
```
Also . SCCACHE link is to oneDNN. please fix.

File: README.md
Comment (skyline75489) [reply README.md:254]: The minimum supported version is 3.6.2, according to setup.py. I added a series of recommended Python version in the Linux guide in #47593 to fix an issue related to anaconda.

File: README.md
Comment (peterjc123) [inline README.md:240]: Cuda -> CUDA

File: README.md
Comment (peterjc123) [inline README.md:242]: are -> is

Comment (peterjc123) [reply README.md:242]: Cuda -> CUDA

File: README.md
Comment (peterjc123) [inline README.md:244]: Be sure -> Make sure

File: README.md
Comment (peterjc123) [inline README.md:252]: It will be great if you could figure out the lower bound for PyTorch version.

File: README.md
Comment (peterjc123) [inline README.md:250]: Pytorch -> PyTorch

File: README.md
Comment (peterjc123) [inline README.md:271]: We may omit this line since CUDA don't force the subversions starting from 10.1. But we may tell the user that it is possible to do that using this variable.

File: README.md
Comment (peterjc123) [inline README.md:282]: It is better to place this paragraph in a better place. Usually users won't read to the end.

File: README.md
Comment (peterjc123) [inline README.md:238]: faily -> fairly
Also sentences starting with repeated `it is` are not good.

Comment (peterjc123) [reply README.md:238]:
```suggestion
It's fairly easy to build with CPU. Visual Studio 2019 version 16.7.6 (MSVC toolchain version 14.27) or higher is recommended.
```

**Review by peterjc123 — COMMENTED** [review]
Some trivial comments

File: README.md
Comment (mszhanyi) [reply README.md:254]: since the suggestion python version is mentioned at the beginning of https://github.com/pytorch/pytorch#from-source and it's platform agnostic. I think we might not add it once more.

File: README.md
Comment (mszhanyi) [reply README.md:258]: Does the previous one refer to those additional libraries, Magma, oneDNN and sccache?
Not like Linux, besides those additional libs, users have to set CUDA_PATH, CUDNN_LIB_DIR, and other envs in windows.
If they uses Sccache, they have to set some [special envs](https://github.com/pytorch/pytorch/blob/8ff0b6fef822edae1b557649b09dccfdaa52a2e3/.jenkins/pytorch/win-test-helpers/build_pytorch.bat#L94-L110) to avoid intermittent [build error](https://github.com/pytorch/pytorch/issues/25393).

File: README.md
Comment (mszhanyi) [reply README.md:252]: good suggestion. Do you know the exact lower bound?

File: README.md
Comment (mszhanyi) [reply README.md:256]: Good catch!

File: README.md
Comment (peterjc123) [reply README.md:252]: I guess you may figure it out by taking a look at `http://download.pytorch.org/whl/{cu92/cu100/cu101}/torch_stable.html`.

File: README.md
Comment (mszhanyi) [reply README.md:275]: I'm a little afraid it might confuse users since it's too detail. Let me think how to make it clear and simple

**Review by walterddr — APPROVED** [review]
looks good to me overall. please feel free to continue refine. I will wait for one more pytorch dev infra member approval to merge.

File: README.md
Comment (mszhanyi) [reply README.md:276]: since vcvarsall.bat needs the variable， shall we keep it？

Comment (peterjc123) [reply README.md:276]: I'm okay with that.

File: README.md
Comment (janeyx99) [inline README.md:247]:
```suggestion
<br/> If Ninja is selected as the generator, the latest MSVC will get selected as the underlying toolchain.
```

File: README.md
Comment (janeyx99) [inline README.md:249]:
```suggestion
MSVC has version dependencies with CUDA and PyTorch, so please install matching versions. 
```

File: README.md
Comment (janeyx99) [inline README.md:257]:
```suggestion
Note: There's a [compilation issue](https://github.com/oneapi-src/oneDNN/issues/812) in serveral Visual Studio 2019 versions since 16.7.1, so please make sure your Visual Studio 2019 version is not in 16.7.1 ~ 16.7.5.
```

File: README.md
Comment (janeyx99) [inline README.md:260]:
```suggestion
[Magma](https://developer.nvidia.com/magma), [oneDNN, a.k.a MKLDNN or DNNL](https://github.com/oneapi-src/oneDNN), and [Sccache](https://github.com/mozilla/sccache). Please refer to the [installation-helper](https://github.com/pytorch/pytorch/tree/master/.jenkins/pytorch/win-test-helpers/installation-helpers) to install them.
```

File: README.md
Comment (samestep) [inline README.md:242]:
```suggestion
[NVTX](https://docs.nvidia.com/gameworks/content/gameworkslibrary/nvtx/nvidia_tools_extension_library_nvtx.htm) is needed to build Pytorch with CUDA.
```

**Review by janeyx99 — COMMENTED** [review]
Looks good, just made some rephrasing nits.

File: README.md
Comment (samestep) [inline README.md:249]:
```suggestion
CUDA, MSVC, and PyTorch versions are interdependent; please install matching versions from this table:
```

File: README.md
Comment (samestep) [inline README.md:259]:
```suggestion
Additional libraries such as
```

File: README.md
Comment (samestep) [inline README.md:262]:
```suggestion
You can refer to the [build_pytorch.bat](https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/win-test-helpers/build_pytorch.bat) script for some other environment variable configurations.
```

File: README.md
Comment (samestep) [inline README.md:268]:
```suggestion
:: [Optional] If you want to build with the VS 2017 generator for old CUDA and PyTorch, please change the value in the next line to `Visual Studio 15 2017`.
```

**Review by samestep — CHANGES_REQUESTED** [review]

**Comment by mszhanyi** [comment]
@walterddr @samestep @janeyx99 @peterjc123  I really appreciate your corrections. It's very important to me

File: README.md
Comment (mszhanyi) [reply README.md:260]: repeated one is deleted.

**Comment by walterddr** [comment]
thanks for the updates, lgtm. merging

**Pull request closed without merging**
